analyses/malawi/dryspells_trigger/07_ecmwf_monthly_skill_general.md

```diff
@@ -41,164 +37,372 @@ reload(processing)
 
 mpl.rcParams['figure.dpi'] = 200
 pd.options.mode.chained_assignment = None
-font = {'family' : 'normal',
-        'weight' : 'normal',
-        'size'   : 16}
+font = {'size'   : 16}
 
 mpl.rc('font', **font)
 ```
 
+```python
+#set plot colors
+hdx_blue='#66B0EC'
+```
+
 #### Set config values
 
 ```python
 country="malawi"
 config=Config()
 parameters = config.parameters(country)
 country_iso3=parameters["iso3_code"]
-
-country_data_raw_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR, config.RAW_DIR,country_iso3)
-country_data_processed_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,config.PROCESSED_DIR,country_iso3)
 country_data_exploration_dir = os.path.join(config.DATA_DIR,config.PUBLIC_DIR,"exploration",country_iso3)
-chirps_country_data_exploration_dir= os.path.join(config.DATA_DIR,config.PUBLIC_DIR, "exploration", country_iso3,'chirps')
+chirps_country_data_exploration_dir= os.path.join(country_data_exploration_dir,'chirps')
+monthly_precip_exploration_dir=os.path.join(country_data_exploration_dir,"dryspells","monthly_precipitation")
 
 chirps_monthly_mwi_path=os.path.join(chirps_country_data_exploration_dir,"chirps_mwi_monthly.nc")
-ecmwf_country_data_processed_dir = os.path.join(country_data_processed_dir,"ecmwf")
-monthly_precip_exploration_dir=os.path.join(country_data_exploration_dir,"dryspells","monthly_precipitation")
+crps_path=os.path.join(monthly_precip_exploration_dir,"mwi_crps.csv")
+```
 
-plots_dir=os.path.join(country_data_processed_dir,"plots","dry_spells")
-plots_seasonal_dir=os.path.join(plots_dir,"seasonal")
+### Read in forecast and observational data
 
-adm2_bound_path=os.path.join(country_data_raw_dir,config.SHAPEFILE_DIR,parameters["path_admin2_shp"])
-all_dry_spells_list_path=os.path.join(country_data_processed_dir,"dry_spells","full_list_dry_spells.csv")
-monthly_precip_path=os.path.join(country_data_processed_dir,"chirps","seasonal","chirps_monthly_total_precipitation_admin1.csv")
+```python
+da_lt=processing.get_ecmwf_forecast_by_leadtime("mwi")
 ```
 
-# Determine general skill per leadtime
+```python
+da_obs=xr.load_dataset(chirps_monthly_mwi_path)
+da_obs=da_obs.precip
+```
 
+```python
+#interpolate forecast data such that it has the same resolution as the observed values
+#using "nearest" as interpolation method and not "linear" because the forecasts are designed to have sharp edged and not be smoothed
+da_forecast=da_lt.interp(latitude=da_obs["latitude"],longitude=da_obs["longitude"],method="nearest")
```

Comment:
What if instead of interpolating the forecast, you down-sampled the observations?

Reply:
Hmm nice idea. The only thing being that down-sampling the observations results in quite some NaN's due to the way the observational data is clipped.. What would you say is the advantage of down-sampling instead of up-sampling?

Reply:
Good idea! My reasoning for down-sampling instead of upsampling would be that the forecast can only predict at the resolution it has, so if comparing with a higher resolution it is unfair cause it will always be off for some parts then. Would you agree on that?

Reply:
Yes agreed, it will give you a better picture of what to expect for the trigger since ultimately you will be using the forecast at the given resolution.

Also, when you interpolate it isn't really the same as increasing the resolution; there are effects to keep in mind, like you will be more likely be missing extreme values in the interpolated dataset.

Reply:
(Oh and it will also make it way less memory intensive to run!)